conans/client/build/meson.py

```diff
@@ -122,6 +125,89 @@ def _get_dirs(self, source_folder, build_folder, source_dir, build_dir, cache_bu
     def flags(self):
         return defs_to_string(self.options)
 
+    def _configure_cross_compile(self, cross_filename, environ_append):
+        cpu_translate = {
+            'armv8': ('aarch64', 'aarch64', 'little'),
+            'x86': ('x86', 'x86', 'little'),
+            'x86_64': ('x86_64', 'x86_64', 'little'),
+        }
+        if hasattr(self._conanfile,'settings_build'):
+            build_cpu_family,build_cpu ,build_endian = cpu_translate[str(self._conanfile.settings_build.arch)]
+            os_build = str(self._conanfile.settings_build.os_build).lower()
```

Comment:
use **get_build_os_arch** to satisfy both old and new cross-building models

Reply:
done